### ISSUE TITLE
datadog_authn_mapping apply error

### ISSUE THREAD

**Issue by nicoodle** [DataDog/terraform-provider-datadog#1385]
Trying to use the new datadog_authn_mapping resource to create SAML groups to Role mappings. All mappings appear correctly in Datadog but the Terraform apply exits with errors. 

### Terraform Version
```
Terraform v1.0.11
on darwin_arm64
+ provider registry.terraform.io/datadog/datadog v3.9.0
```

### Affected Resource(s)
-   datadog_authn_mapping

### Terraform Configuration Files
```hcl
---tfvars---

role_mappings = {
  "Datadog Admin Role"       = "AZUREAD_GROUP_ID" # Datadog-Admins
  "Datadog Production Role"  = "AZUREAD_GROUP_ID" # Datadog-ProductionUsers
  "Datadog Development Role" = "AZUREAD_GROUP_ID" # Datadog-DevelopmentUsers
  "Datadog Read Only Role"   = "AZUREAD_GROUP_ID" # Datadog-ReadOnly
}
```

```hcl
data "datadog_role" "roles" {
  for_each = var.role_mappings
  filter   = each.key
  depends_on = [
    datadog_role.development_role,
    datadog_role.production_role
  ]
}

resource "datadog_authn_mapping" "mappings" {
  for_each = var.role_mappings
  key      = "groups"
  value    = each.value
  role     = data.datadog_role.roles[each.key].id
}
```

### Expected Behavior

Role mappings apply without error.

### Actual Behavior

Role mappings appear within Datadog platform but Terraform errors with the following:

```
│ Error: object contains unparsed element: map[role:map[data:map[id:507f6b72-a05a-11ec-b088-da7ad0900005 type:roles]] saml_assertion_attribute:map[data:map[id:6027 type:saml_assertion_attributes]]]
│
│   with datadog_authn_mapping.mappings["Datadog Development Role"],
│   on permissions.tf line 36, in resource "datadog_authn_mapping" "mappings":
│   36: resource "datadog_authn_mapping" "mappings" {
│
```

### Steps to Reproduce

`terraform apply`

**Comment by skarimo** [comment]
I have trying to reproduce this but am unable to. Could you forward debug logs from terraform using `TF_LOG=debug`. These logs should contain the exact payloads for me to reproduce. Please redact any sensitive information in the logs.

**Comment by dawiki** [comment]
I think that the overall problem here is that the DataDog API is not behaving according to it's API spec: https://docs.datadoghq.com/api/latest/authn-mappings/#get-an-authn-mapping-by-uuid

Example response with replaced ids:
```

{
  "included": [
    {
      "type": "saml_assertion_attributes",
      "id": 1234,
      "attributes": {
        "attribute_key": "member-of",
        "attribute_value": "owner"
      }
    },
    {
      "type": "roles",
      "id": "11111111-2222-3333-aaaa-bbbbbbbbbbbb",
      "attributes": {
        "name": "Datadog Admin Role",
        "created_at": "2022-05-11T06:38:18.140457+00:00",
        "modified_at": "2022-05-11T06:38:18.140457+00:00"
      },
      "relationships": {
        "permissions": {
          "data": [
            {
              "type": "permissions",
              "id": "11111111-2222-3333-aaaa-bbbbbbbbbbbb"
            },
            ...
          ]
        }
      }
    }
  ],
  "data": {
    "type": "authn_mappings",
    "id": "11111111-2222-3333-aaaa-bbbbbbbbbbbb",
    "attributes": {
      "saml_assertion_attribute_id": 1234,
      "attribute_key": "member-of",
      "attribute_value": "owner",
      "created_at": "2022-05-12T09:15:55.512092+00:00",
      "modified_at": "2022-05-12T11:45:54.142468+00:00"
    },
    "relationships": {
      "saml_assertion_attribute": {
        "data": {
          "type": "saml_assertion_attributes",
          "id": "1234"
        }
      },
      "role": {
        "data": {
          "type": "roles",
          "id": "11111111-2222-3333-aaaa-bbbbbbbbbbbb"
        }
      }
    }
  }
}
```
Expected response structure
```
{
  "data": {
    "type": ...,
    "id": ...,
    "attributes": {
      ...
    },
    "relationships": {
      "saml_assertion_attribute": {
        "data": {
          ...
        }
      },
      "role": {
        "data": {
          ...
        }
      }
    },
    "included": [
      {
        "type": ...,
        "id": ...,
        "attributes": {
          ...
        }
      },
      ...
    }
    ]
  }
}
```

Notice how the the `included` attribute is next to `data` when it should be one level lower. I think this is a problem on the authn endpoint side since all other occurrences I found (including the documentation and spec, provided and generated api clients) are aligned. I would appreciate it if you could fix this and I am stunned that nobody (or no contract test) caught this already.

**Comment by skarimo** [comment]
This should be resolved in the latest provider release v3.13.1.

**Comment by tobiaswi** [comment]
A bit more context for everyone here:
If you tried the v.3.13.1 provider before 19/07/22 your apply would fail. There was a backend fix deployed yesterday and the same, unchanged provider now applies without errors:
![image](https://user-images.githubusercontent.com/15244405/179913245-eb3680dd-1296-426d-8eca-5c12146fb84d.png)

Thanks dd team, this is such an important fix for us.
